constellations/orion/src/controllers/classes_by_semester_controller_test.go

```diff
@@ -0,0 +1,80 @@
+package controllers_test
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/controllers/testUtils"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/domains"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/repos"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test getting programs and classes by semester endpoint
+func TestGetClassesAndProgramsBySemester_Success(t *testing.T) {
+	// Mock selecting all programs, semesters, classes
+	testUtils.ProgramRepo.MockSelectAll = func(publishedOnly bool) ([]domains.Program, error) {
+		return []domains.Program{
+			{
+				Id:          1,
+				ProgramId:   "ap_calculus",
+				Name:        "AP Calculus",
+				Grade1:      2,
+				Grade2:      3,
+				Description: "Description1",
+				Featured:    0,
+			},
+			{
+				Id:          2,
+				ProgramId:   "ap_java",
+				Name:        "AP Java",
+				Grade1:      2,
+				Grade2:      3,
+				Description: "Description2",
+				Featured:    1,
+			},
+		}, nil
+	}
+	repos.ProgramRepo = &testUtils.ProgramRepo
```

Comment:
Becuase we'll have multiple tests, this part is going to be repeated a lot.Is there a way we make this easier on ourselves and simply have a function that looks like this:

`mockPrograms(...)` which we call in every test. so it shrinks the amount of code per test?